x-pack/legacy/plugins/maps/public/components/gis_map/view.js

```diff
@@ -57,8 +68,22 @@ export class GisMap extends Component {
     }
   };
 
+  // Mapbox does not provide any feedback when rendering is complete.
+  // Temporary solution is just to wait set period of time after data has loaded.
+  startRenderTimer = () => {
+    setTimeout(
```

Comment:
I understand this event is really only relevant once downstream, because we only listen to it once for a single report.

But conceptually it makes less sense, when this code is not viewed in that context.

I would:
- set `state.isRenderTimeoutComplete` to false at the beginning of each componentDidUpdate.
- keep track of the timeoutHandle and `clearTimeout` when the component unmounts. (similar to the interval-timer) (this var could then replace the `_isRenderTimerStarted` too)

Reply:
otherwise, rename renderTimeout to something along the lines of `initialLoadTimeout`

Reply:
I will rename `isRenderTimeoutComplete ` to `isInitialLoadRenderTimeoutComplete` since `data-render-complete` is only used by reporting and only cares about the initial render.